File: tests/src/main/java/com/github/klee0kai/test/tech/phone/base/LifecycleUtils.java
package com.github.klee0kai.test.tech.phone.base;

import com.github.klee0kai.stone.types.lifecycle.StoneLifeCycleOwner;

public class LifecycleUtils {

    public static StoneLifeCycleOwner createFromATech(ATech phone){
        return listener -> phone.subscribe(
                new ATechLifecycle() {
                    @Override
                    public void onBuy() {

                    }

                    @Override
                    public void onBroken() {

                    }

                    @Override
                    public void onDrown() {
                        listener.protectForInjected(100);
                    }
                }
        );
    }


}

File: tests/src/commonMain/kotlin/com/github/klee0kai/test/tech/phone/base/LifecycleUtils.kt
package com.github.klee0kai.test.tech.phone.base

import com.github.klee0kai.stone.lifecycle.StoneLifeCycleListener
import com.github.klee0kai.stone.lifecycle.StoneLifeCycleOwner

object LifecycleUtils {
    fun createFromATech(phone: ATech): StoneLifeCycleOwner {
        return StoneLifeCycleOwner { listener: StoneLifeCycleListener? ->
            phone.subscribe(
                object : ATechLifecycle {
                    override fun onBuy() {
                    }

                    override fun onBroken() {
                    }

                    override fun onDrown() {
                        listener!!.protectForInjected(100)
                    }
                }
            )
        }
    }
}
